lib/src/model/robotoff_nutrient_extraction_annotation.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:json_annotation/json_annotation.dart';
+import 'package:openfoodfacts/src/model/nutrient_modifier.dart';
+import 'package:openfoodfacts/src/model/per_size.dart';
+import 'package:openfoodfacts/src/utils/unit_helper.dart';
+
+part 'robotoff_nutrient_extraction_annotation.g.dart';
+
+@JsonSerializable()
+class RobotoffNutrientAnnotationData {
+  @JsonKey(toJson: UnitHelper.unitToString, fromJson: UnitHelper.stringToUnit)
+  final Unit? unit;
+  final String value;
+  final NutrientModifier? modifier;
+  final double? numericalValue;
+
+  RobotoffNutrientAnnotationData({
+    this.unit,
+    required this.value,
+  })  : modifier = NutrientModifierExtension.fromValue(value),
+        numericalValue = double.tryParse(value.trim().substring(1));
```

Comment:
What if there's no modifier? Still starting at 1?
What if the string is empty?
Besides, getters would be good enough, instead of final fields.